gradoop-data-integration/src/main/java/org/gradoop/dataintegration/importer/csv/MinimalCSVImporter.java

```diff
@@ -0,0 +1,169 @@
+   */
+  private List<String> columnNames;
+
+  /**
+   * Gradoop Flink configuration
+   */
+  private GradoopFlinkConfig config;
+
+  /**
+   * Create a new MinimalCSVImporter. The user set a list of the property names.
+   * @param path the path to the csv file
+   * @param tokenSeperator the token delimiter of the csv file
+   * @param config GradoopFlinkConfig
+   * @param columnNames property identifier for each column
+   */
+  public MinimalCSVImporter(String path, String tokenSeperator, GradoopFlinkConfig config,
+          List<String> columnNames) {
+    this(path, tokenSeperator, config);
+    this.columnNames = columnNames;
+  }
+
+  /**
+   * Create a new MinimalCSVImporter. Use the utf-8 charset as default. The first line of the file
+   * will set as the property names for each column.
+   * @param path the path to the csv file
+   * @param tokenSeperator the token delimiter of the csv file
+   * @param config GradoopFlinkConfig
+   */
+  public MinimalCSVImporter(String path, String tokenSeperator, GradoopFlinkConfig config) {
+    this(path, tokenSeperator, config, "UTF-8");
+  }
+
+  /**
+   * Create a new MinimalCSVImporter with a user set charset. The first line of the file
+   * will set as the property names for each column.
+   * @param path the path to the csv file
+   * @param tokenSeperator the token delimiter of the csv file
+   * @param config GradoopFlinkConfig
+   * @param charset the charset used in the csv file
+   */
+  public MinimalCSVImporter(String path, String tokenSeperator, GradoopFlinkConfig config,
+          String charset) {
+    this.path = path;
+    this.tokenSeparator = tokenSeperator;
+    this.config = config;
+    this.charset = charset;
+  }
+
+  /**
+   * Import each row of the file as a vertex. If no column property names are set,
+   * read the first line of the file as header and set this values as column names.
+   * @param checkReoccurringHeader if each row of the file should be checked for reocurring of
+   * the column property names.
+   * @return the imported vertices
+ * @throws IOException if an error occurred while open the stream
+   */
+  public DataSet<ImportVertex<Long>> importVertices(boolean checkReoccurringHeader)
+          throws IOException {
+    if (columnNames == null) {
+      return readCSVFile(readHeaderRow(), checkReoccurringHeader);
+    } else {
+      return readCSVFile(columnNames, checkReoccurringHeader);
+    }
+  }
+
+  /**
+   * Read the vertices from a csv file.
+   *
+   * @param propertyNames list of the property identifier name
+   * @param checkReoccurringHeader if each row of the file should be checked for reocurring of
+   * the column property names.
+   * @return DateSet of all vertices from one specific file.
+   */
+  public DataSet<ImportVertex<Long>> readCSVFile(List<String> propertyNames,
+          boolean checkReoccurringHeader) {
+
+    DataSet<Properties> lines = config.getExecutionEnvironment()
+    .readTextFile(path)
```

Comment:
indentation ... same for lines below
```suggestion
      .readTextFile(path)
```